### PR TITLE
Ruby on Rails capstone project - E-nstaPay Budget App

### PR DESCRIPTION
**Pull request by ProgramKingAbel**
# E-nstaPay Budgeting application :1234: 

## Technical Implementation :hammer_and_pick: 

- Used Postgres as your database.
- Used devise for authentication.
- Used .erb view template 
- The project  is [deployed and accessible online](https://instapay-z93h.onrender.com) on render
- The database schema reflects the structure on the ERD Diagram

## Interactions :pinched_fingers: 

- Splash screen with and links to the sign up and log in pages.
- Sign up and log in pages
- Home page (categories page)
- Transactions page
- "Add a new category" page
- "Add a new transaction" page

## Testing - Unit and Integration :test_tube: 
- Unit testing with RSpec
- Integration tests with Capybara

### CODE REVIEW

File: app/views/payments/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      
+      
+    </tr>
+    <% end %>
+  </tbody>
+</table>
+
+<%= link_to 'Add a New Transaction', new_group_payment_path(@group), class: "btn btn-primary mt-2" %>
+
+</div>
```
Comment (Whoistolu) [inline app/views/payments/index.html.erb]: - [ ] [REQUIRED] You have done a great job so far @ProgramKingAbel , however, it is a requirement to abide with the design template  provided in terms of colour,and the general layout. Let's make sure that the button to add transactions is green and fixed at the bottom of the screen just as it it in the design template provided.

Your Implementation | Template Design
:--------------------: | :-------------------:
Screenshot 2023-09-29 at 22 06 16 | Screenshot 2023-09-29 at 22 05 34

File: app/views/groups/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+</div>
+
+<div class="m-2">
+
+  <%= f.label :icon %>
+  <%= f.text_field :icon, autofocus: true, autocomplete: "icon", class: "form-control", placeholder: "Enter icon url"  %>
+</div>
+  <%= f.submit 'Create Category', class: "btn btn-primary add-new m-2" %>
+<% end %>
+</div>
```
Comment (Whoistolu) [inline app/views/groups/new.html.erb:23]: - [ ] [REQUIRED] Just as I have recommended on the transaction page, kindly implement the same for the category page as well. Let's ensure that the button colour is adjusted and the button position is fixed beneath the screen as it is in the provided design template.

- [ ] [REQUIRED] You have done a great job so far @ProgramKingAbel , however, it is a requirement to abide with the design template  provided in terms of colour,and the general layout. Let's make sure that the button to add transactions is green and fixed at the bottom of the screen just as it it in the design template provided.

Your Implementation | Template Design
:--------------------: | :-------------------:
Screenshot 2023-09-29 at 22 26 32 | Screenshot 2023-09-29 at 22 05 34

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @ProgramKingAbel 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### Highlight
🔆 Devise is properly installed
🔆 A user can sign up and sign in successfully
🔆 When the user logs in, they are presented with the categories page
🔆 For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category
🔆 For a given category, the list of transactions is presented ordered by the most recent
🔆 There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction
🔆 When the user clicks on the "Back" button (<), the user navigates to the home page

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @ProgramKingAbel, 
congrats 2

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**